### PR TITLE
Fix: hotfix for get user with no roles

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

 hotfix for no roles found on get user

Fixes # (issue) : https://github.com/devtron-labs/devtron/issues/1329

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x]  get user for system

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

File: pkg/user/UserService.go [hunk 1]
```diff
@@ -676,14 +676,19 @@ func (impl UserServiceImpl) GetById(id int32) (*bean.UserInfo, error) {
...
+		filterGroupsModels, err := impl.roleGroupRepository.GetRoleGroupListByCasbinNames(filterGroups)
+		if err != nil {
+			impl.logger.Warnw("No Roles Found for user", "id", model.Id)
+		}
+		filterGroups = nil
+		for _, item := range filterGroupsModels {
+			filterGroups = append(filterGroups, item.Name)
+		}
+	} else {
+		impl.logger.Infow("no roles found for user", "email", model.EmailId)
```
Comment (manish-agrawal-ai) [inline pkg/user/UserService.go:689]: Warnw

File: pkg/user/UserService.go [hunk 2]
```diff
@@ -676,14 +676,19 @@ func (impl UserServiceImpl) GetById(id int32) (*bean.UserInfo, error) {
...
-	if err != nil {
-		impl.logger.Warnw("No Roles Found for user", "id", model.Id)
-	}
-	filterGroups = nil
-	for _, item := range filterGroupsModels {
-		filterGroups = append(filterGroups, item.Name)
+	if len(filterGroups) > 0 {
+		filterGroupsModels, err := impl.roleGroupRepository.GetRoleGroupListByCasbinNames(filterGroups)
+		if err != nil {
+			impl.logger.Warnw("No Roles Found for user", "id", model.Id)
```
Comment (manish-agrawal-ai) [inline pkg/user/UserService.go:682]: Aren't we handling for err != nil case?

File: pkg/user/UserService.go [hunk 3]
```diff
@@ -676,14 +676,19 @@ func (impl UserServiceImpl) GetById(id int32) (*bean.UserInfo, error) {
...
+		}
+		filterGroups = nil
```
Comment (manish-agrawal-ai) [inline pkg/user/UserService.go:684]: initialise with nil or empty? Does this have any impact?

File: pkg/user/UserService.go (hunk 1 above)
Comment (vikramdevtron) [reply pkg/user/UserService.go]: done

**Review by manish-agrawal-ai — APPROVED** [review]

**Pull request merged**